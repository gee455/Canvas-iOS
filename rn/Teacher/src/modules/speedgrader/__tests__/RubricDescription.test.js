//
// Copyright (C) 2016-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

// @flow

import React from 'react'
import { RubricDescription, mapStateToProps } from '../RubricDescription'
import renderer from 'react-test-renderer'

jest.unmock('ScrollView')

const templates = {
  ...require('../../../__templates__/helm'),
  ...require('../../../redux/__templates__/app-state'),
  ...require('../../../__templates__/rubric'),
  ...require('../../../__templates__/assignments'),
  ...require('../../../__templates__/session'),
}

let ownProps = {
  assignmentID: '1',
  rubricID: '1',
  navigator: templates.navigator(),
}

let defaultProps = {
  ...ownProps,
  description: 'A satisfactory description',
}

describe('RubricDescription', () => {
  beforeEach(() => jest.resetAllMocks())

  it('renders', () => {
    let tree = renderer.create(
      <RubricDescription {...defaultProps} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders without description', () => {
    let props = {
      ...defaultProps,
      description: null,
    }
    let tree = renderer.create(
      <RubricDescription {...props} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('mapStateToProps', () => {
  it('returns an empty description when there is no rubric', () => {
    let state = templates.appState({
      entities: {
        assignments: {
          '1': {
            data: templates.assignment({
              id: '1',
            }),
          },
        },
      },
    })

    let props = mapStateToProps(state, ownProps)
    expect(props.description).toEqual('')
  })

  it('returns the rubric description whent here is a rubric', () => {
    let rubric = templates.rubric({ id: '1' })
    let state = templates.appState({
      entities: {
        assignments: {
          '1': {
            data: templates.assignment({
              id: '1',
              rubric: [rubric],
            }),
          },
        },
      },
    })

    let props = mapStateToProps(state, ownProps)
    expect(props.description).toEqual(rubric.long_description)
  })
})
