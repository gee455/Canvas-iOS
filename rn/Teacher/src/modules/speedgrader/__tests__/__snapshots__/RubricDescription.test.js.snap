// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RubricDescription renders 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 16,
        "paddingTop": 24,
      }
    }
  >
    <View
      style={undefined}
      testID="web-container.view"
    >
      <CanvasWebView
        automaticallySetHeight={true}
        bounces={false}
        html="A satisfactory description"
        navigator={
          Object {
            "dismiss": [Function],
            "dismissAllModals": [Function],
            "isModal": false,
            "pop": [Function],
            "show": [Function],
            "traitCollection": [Function],
          }
        }
        onError={[Function]}
        onFinishedLoading={[Function]}
        onHeightChange={[Function]}
        onMessage={[Function]}
        onNavigation={[Function]}
        scrollEnabled={true}
        source={
          Object {
            "baseURL": "http://mobiledev.instructure.com/",
            "html": "A satisfactory description",
          }
        }
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`RubricDescription renders without description 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 32,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#2D3B45",
          "fontSize": 16,
          "lineHeight": 19,
          "textAlign": "center",
        }
      }
    >
      There currently is no long description for this item.
    </Text>
  </View>
</RCTSafeAreaView>
`;
